### DISCUSSION TITLE
ravendb-go-client / context

### DISCUSSION THREAD

**Discussion by felipecrrmts** [ravendb/ravendb#13871] [Q&A]
Are there a ravendb-go-client implementation that uses [context package](https://pkg.go.dev/context) ?

**Comment by DejanMilicic** [comment]
We are not supporting it at the moment. I checked implementation and it should not be a problem, so I will create an issue for this.

**Reply by ayende** [reply]
To give more context, we need to change it here:

https://github.com/ravendb/ravendb-go-client/blob/master/request_executor.go#L418

At the time we wrote the client, `NewRequestWithContext` wasn't available.

The issue is what is the level of the context. I suppose it would make the most sense to define that at the session level.
